scalafmt-core/shared/src/main/scala/org/scalafmt/config/SortSettings.scala

```diff
@@ -1,24 +1,34 @@
 package org.scalafmt.config
 
 import metaconfig._
+import scala.meta.classifiers.Classifier
+import scala.meta.Tree
+import scala.meta.Mod
+import sourcecode.Name
+import sourcecode.Text
 
 case class SortSettings(
     order: List[SortSettings.ModKey]
 )
 
 object SortSettings {
-
-  implicit val SortSettingsModKeyReader: ConfCodec[ModKey] =
-    ReaderUtil.oneOf[ModKey](
-      `implicit`,
-      `final`,
-      `sealed`,
-      `abstract`,
-      `override`,
-      `private`,
-      `protected`,
-      `lazy`
-    )
+  final case class ModKey(
+      name: String,
+      matches: Mod => Boolean
+  ) extends Product
+  private def createMod[T <: Mod](implicit
+      name: Name,
+      c: Classifier[Tree, T]
+  ): ModKey = ModKey(name.value, m => c(m))
```

Comment:
why not `c.apply`?

Reply:
No specific reason. I admit this method was unnecessarily complicated so I removed the method.